app/javascript/components/timeline-options/timeline-helper.js

```diff
@@ -0,0 +1,115 @@
+export const timelineUiOptions = (title) => ({
+  title,
+  axes: {
+    bottom: {
+      title: __('Date'),
+      mapsTo: 'date',
+      scaleType: 'time',
+    },
+    left: {
+      title: __('# of Events'),
+      mapsTo: 'value',
+      scaleType: 'linear',
+    },
+  },
+  legend: {
+    clickable: true,
+  },
+  tooltip: {
+    totalLabel: __('Total Events'),
+  },
+  points: {
+    radius: 3,
+    fillOpacity: 1,
+    filled: true,
+    enabled: true,
+  },
+  zoomBar: {
+    top: {
+      className: 'zoom-bar',
+      enabled: true,
+      type: 'graph_view',
+    },
+  },
+  height: '400px',
+});
+
+const smartAndOrStatements = (group, array) => {
+  const item = (array.length === 1) ? array[0] : `[${array.toString()}]`;
+  return `&filter[]=${group}=${item}`;
+};
+
+/** Function to build the URL to get all events that fall under the user selected parameters. */
+export const buildUrl = (values, pageValues) => {
+  // TODO: Different timeline show different data, ensure all necesary data is pulled
+  let url = `/api/event_streams?limit=5000&offset=0&expand=resources`;
+  url += `&attributes=group,group_level,group_name,id,event_type,ems_id,type,timestamp,created_on,host.name,source,host_id`;
+
+  // Page / Record set values
+  url += `&filter[]=ems_id=${pageValues.ems_id}`;
+  url += pageValues.host_id ? `&filter[]=host_id=${pageValues.host_id}` : ``;
```

Comment:
We probably need to check if `pageValues.ems_id` is defined.  It looks like it's being set regardless.

I'm not sure why `pageValues.host_id` is not defined in the case you're on the host screen... see problem 4) in my comments.

Reply:
> check if pageValues.ems_id is defined

This is my bad, some changes i made to `app/views/layouts/_tl_show.html.haml` were not pushed up. Fixing now. 

> pageValues.host_id is not defined in the case you're on the host screen 

This is because we're pulling the values from `@record` which can have different types and thus different values. An InfraManager will have `ems_id` and `host_id` but a Host wont have `host_id` because the "host id" is just `id`.